backend/src/main/java/de/rgmcode/backend/Security/SecurityConfig.java

```diff
@@ -0,0 +1,50 @@
+package de.rgmcode.backend.Security;
+
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
+
+/*
+@EnableWebSecurity
+@Configuration
+public class SecurityConfig {
+
+    @Bean
+    public PasswordEncoder passwordEncoder(){
+//        return NoOpPasswordEncoder.getInstance();
+        return Argon2PasswordEncoder.defaultsForSpringSecurity_v5_8();
+    }
+
+    @Bean
+    public SecurityFilterChain filterChain(HttpSecurity http) throws Exception {
+        CsrfTokenRequestAttributeHandler requestHandler = new CsrfTokenRequestAttributeHandler();
+        requestHandler.setCsrfRequestAttributeName(null);
+        return http
+                .csrf((csrf)-> csrf
+                        .csrfTokenRepository(CookieCsrfTokenRepository.withHttpOnlyFalse())
+                        .csrfTokenRequestHandler(requestHandler))
+                .httpBasic(Customizer.withDefaults())
+//                .sessionManagement(httpSecuritySessionManagementConfigurer ->
+//                        httpSecuritySessionManagementConfigurer.sessionCreationPolicy(SessionCreationPolicy.ALWAYS))
+                // popups im browser unterdrücken
+                .sessionManagement(session -> session.sessionCreationPolicy(SessionCreationPolicy.ALWAYS))
+                .httpBasic(basic -> basic.authenticationEntryPoint(
+                        (request, response, authException) ->
+                                response.sendError(
+                                        HttpStatus.UNAUTHORIZED.value(),
+                                        HttpStatus.UNAUTHORIZED.getReasonPhrase()
+                                )))
+                .authorizeHttpRequests(httpRequests ->
+                                httpRequests
+                                        .requestMatchers(HttpMethod.POST, "/api/user").permitAll()
+                                        .requestMatchers("/api/user/**").permitAll()
+                                        // .requestMatchers("/api/employees").hasRole("ADMIN") -> User Roles anlegen!!
+                                        .anyRequest()
+                                        .permitAll()
+//                                .authenticated()
+                )
+                .build();
+    }
+
+}*/
```

Comment:
Here is a lot of commented code. Is this even working? If you don't need it, don't put it in your "production" code ;-)

Reply:
I had pre-committed the code, actually it shouldn't be committed, I forgot to exclude that.